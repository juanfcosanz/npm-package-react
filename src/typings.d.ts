// * declaraci贸n de m贸dulos para CSS
declare module '*.css' {
  const content: { [className: string]: string };
  export default content;
}

/*
 * * declaraci贸n de m贸dulos para imagenes
 * NOTA: para cada tipo de formato de imagenes debe declarase como un modulo independiente
 */
declare module '*.jpg' {
  const value: any;
  export default value;
}
