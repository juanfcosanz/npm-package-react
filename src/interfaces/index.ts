export * from './cart.interfaces';
export * from './product.interfaces';
