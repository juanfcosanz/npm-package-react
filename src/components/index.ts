import { IProductCardProps } from '../interfaces';

import { ProductCard as ProductCardComponent } from './ProductCard';
import { ProductButtons, ProductImage, ProductTitle } from './ProductCardElement';

export * from './ProductCard';
export * from './ProductCardElement';

// * de esta forma al componente ProductCard le expandimos nuevas propiedades
export const ProductCard: IProductCardProps = Object.assign(ProductCardComponent, {
	Title: ProductTitle,
	Image: ProductImage,
	Buttons: ProductButtons,
});
