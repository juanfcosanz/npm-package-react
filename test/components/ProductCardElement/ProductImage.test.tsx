import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../../src';
import { product2 } from '../../data/products';

describe('ProductImage', () => {
  // ? Nombre de la prueba
  test('Debe de mostrar el componente correctamente con la imagen personalizada', () => {
    //? montamos nuestro componente (renderer es una funcion instalada )
    const wrapper = renderer.create(<ProductImage img={'https://hola.jpg'} />);

    // ? creamos un snapshot de nuestro componente renderizado
    // * cada que cambiemos algo en el wrapper, se evalúa si la foto guardada coincide con la actual
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  // //? cuando no se envía el title, viene del product
  test('Debe de mostrar el componente con la imagen del producto', () => {
    const wrapper = renderer.create(
      // * como el product viene del Context que es proporcionado por el ProductCard
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
