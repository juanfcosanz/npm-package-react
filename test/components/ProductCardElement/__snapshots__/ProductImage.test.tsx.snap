// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe de mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product image"
    className="productImg undefined"
    src="./coffee-mug2.png"
  />
</div>
`;

exports[`ProductImage Debe de mostrar el componente correctamente con la imagen personalizada 1`] = `
<img
  alt="Product image"
  className="productImg undefined"
  src="https://hola.jpg"
/>
`;
